core/src/main/java/org/infinispan/expiration/impl/InternalExpirationManager.java

```diff
@@ -74,14 +74,12 @@ default void handleInStoreExpiration(MarshallableEntry<K, V> marshalledEntry) {
    }
 
    /**
-    * Retrieves the last access time for the given key in the data container if it is using max idle.
-    * If the entry is not in the container or it is expired it will return null.
-    * If the entry is present but cannot expire via max idle, it will return -1
-    * If the entry is present and can expire via max idle but hasn't it will return a number > 0
-    * @param key the key to retrieve the access time for
-    * @param value the value to match if desired (this can be null)
-    * @param segment the segment for the given key
-    * @return the last access time if available
+    * Handles processing for an entry that may be expired. This will remove the entry if it is expired, otherwise may
+    * touch if it uses max idle. The return stage will contain whether the entry was actually expired or not
+    * @param entry entry that may be expired
+    * @param segment the segment of the entry
+    * @param isWrite whether the command that saw the expired value was a write or not
+    * @return a stage that when complete will return if the entry was expired
```

Comment:
I prefer saying `a stage that will complete with {@code true} if the entry was expired and {@code false} otherwise.
Also duplicated in the main comment (`The return stage will contain whether the entry was actually expired or not`)